go/merge/three_way.go

```diff
@@ -63,7 +63,15 @@ func unmergeable(a, b types.Value) bool {
 	return true
 }
 
-func threeWayMerge(a, b, parent types.Value, vwr types.ValueReadWriter) (merged types.Value, err error) {
+func updateProgress(progress chan struct{}) {
+	// TODO: Eventually we'll want more information than a single bit :).
+	if progress != nil {
+		progress <- struct{}{}
+	}
+}
+
+func threeWayMerge(a, b, parent types.Value, vwr types.ValueReadWriter, progress chan struct{}) (merged types.Value, err error) {
+	defer updateProgress(progress)
```

Comment:
Since you're calling this down in all the `apply()` closures, I don't think it's accurate to also call it here. To a first approximation, you want to do this every time the code calls Set or Remove or similar on `merged`, right? That doesn't happen here in this function